src/azure-cli/azure/cli/command_modules/storage/azcopy/util.py

```diff
@@ -30,10 +31,26 @@ class AzCopy:
     def __init__(self, creds=None):
         self.system = platform.system()
         install_location = _get_default_install_location()
-        self.executable = install_location
+        self.executable = None
+        if os.path.isfile(install_location):
+            self.executable = install_location
+        else:
+            try:
+                import re
+                from packaging.version import parse as parse_version
+                args = ["azcopy", "--version"]
+                out_bytes = subprocess.check_output(args)
+                out_text = out_bytes.decode('utf-8')
+                version = re.findall(r"azcopy version (.+?)\n", out_text)[0]
+                if parse_version(version) >= parse_version(AZCOPY_VERSION):
+                    self.executable = "azcopy"
+            except FileNotFoundError:
```

Comment:
If there's other error, will CLI be crashed? Is it possible that other errors be thrown?

Reply:
Using exception to catch all.